systemtests/src/main/java/io/enmasse/systemtest/utils/IoTUtils.java

```diff
@@ -69,15 +96,37 @@ public static void deleteIoTConfigAndWait(Kubernetes kubernetes, IoTConfig confi
     }
 
     private static void waitForIoTConfigDeleted(Kubernetes kubernetes) throws Exception {
-        TestUtils.waitForNReplicas(0, false, IOT_LABELS, Collections.emptyMap(), new TimeoutBudget(2, TimeUnit.MINUTES), 5000);
+        TestUtils.waitForNReplicas(0, false, IOT_LABELS, Collections.emptyMap(), new TimeoutBudget(5, TimeUnit.MINUTES), 5000);
     }
 
-    private static boolean allDeploymentsPresent(Kubernetes kubernetes) {
+    private static boolean allDeploymentsPresent(Kubernetes kubernetes, String[] expectedDeployments) {
         final String[] deployments = kubernetes.listDeployments(IOT_LABELS).stream()
                 .map(deployment -> deployment.getMetadata().getName())
                 .toArray(String[]::new);
         Arrays.sort(deployments);
-        return Arrays.equals(deployments, EXPECTED_DEPLOYMENTS);
+        Arrays.sort(expectedDeployments);
+
+        return Arrays.equals(deployments, expectedDeployments);
+    }
+
+    private static String[] getExpectedDeploymentsNames(IoTConfig config) {
+        List<String> expectedDeployments = new ArrayList<>();
+        addIfEnabled(expectedDeployments, config, AdaptersConfig::getHttp, IOT_HTTP_ADAPTER);
+        addIfEnabled(expectedDeployments, config, AdaptersConfig::getLoraWan, IOT_LORAWAN_ADAPTER);
+        addIfEnabled(expectedDeployments, config, AdaptersConfig::getMqtt, IOT_MQTT_ADAPTER);
+        addIfEnabled(expectedDeployments, config, AdaptersConfig::getSigfox, IOT_SIGFOX_ADAPTER);
+        expectedDeployments.addAll(Arrays.asList(COMPONENTS));
+        return expectedDeployments.toArray(String[]::new);
+    }
+
+    private static void addIfEnabled(List<String> adapters, IoTConfig config, Function<AdaptersConfig, AdapterConfig> adapterGetter, String name) {
+        Optional<AdapterConfig> adapter = Optional.ofNullable(config.getSpec().getAdapters()).map(adapterGetter);
```

Comment:
Just a thought: You could "map" the result even further:

~~~java
 Optional<Boolean> enabled = Optional.ofNullable(config.getSpec().getAdapters()).map(adapterGetter).map(AdapterConfig::getEnabled);
~~~

And then:

~~~java
if(enabled.orElse(true)) {
  …
} else {
  …
}
~~~